static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -61,72 +61,80 @@ define([
-                                'position': '',
-                                'top': '',
-                                'max-height': ''
+                                'position': 'absolute',
+                                'top': state.headerHeight
                             });
+                        } else {
+                            $adBanner.css({ 'position': 'fixed' });
+                        }
 
-                            // Set
-                            $header.css({ 'margin-top': state.adHeight });
-
-                            $adBanner.css({ 'max-height': state.adHeight });
-                            var userHasScrolledPastHeader = scrollY > state.headerHeight;
-                            if (userHasScrolledPastHeader) {
-                                $adBanner.css({
-                                    'position': 'absolute',
-                                    'top': state.headerHeight
-                                });
-                            } else {
-                                $adBanner.css({ 'position': 'fixed' });
-                            }
-
-                            var scrollIsAtTop = scrollY === 0;
-                            var diff = state.adHeight - state.previousAdHeight;
-                            var adHeightHasIncreased = diff > 0;
-                            if (!scrollIsAtTop && adHeightHasIncreased) {
-                                // If the user is not at the top and the ad height has increased,
-                                // we want to offset their scroll position
-                                window.scrollTo(scrollX, scrollY + diff);
-                            } else if (!state.firstRender) {
-                                // Otherwise we want to transition the change when it happens.
-                                // Avoid an initial transition when we apply the margin top for the first time
-                                $header.css('transition', 'margin-top 1s cubic-bezier(0, 0, 0, 0.985)');
-                            }
-                        });
+                        var scrollIsAtTop = state.scrollY === 0;
+                        var diff = state.adHeight - state.previousAdHeight;
+                        var adHeightHasIncreased = diff > 0;
+                        if (!scrollIsAtTop && adHeightHasIncreased) {
+                            // If the user is not at the top and the ad height has increased,
+                            // we want to offset their scroll position
+                            window.scrollTo(state.scrollX, state.scrollY + diff);
+                        } else if (!state.firstRender) {
+                            // Otherwise we want to transition the change when it happens.
+                            // Avoid an initial transition when we apply the margin top for the first time
+                            $header.css('transition', 'margin-top 1s cubic-bezier(0, 0, 0, 0.985)');
```

Comment:
I wonder if it is there is an opportunity to use classes to switch on/off the positioning and transitioning.